File: android/src/main/java/org/ligi/gobandroid_hd/ui/links/LinkListFragment.java
package org.ligi.gobandroid_hd.ui.links;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import org.ligi.gobandroid_hd.R;

public abstract class LinkListFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
        RecyclerView recyclerView = (RecyclerView) inflater.inflate(R.layout.recycler_view, container, false);
        recyclerView.setAdapter(new TwoLineRecyclerAdapter(getData()));
        recyclerView.setLayoutManager(new LinearLayoutManager(container.getContext()));
        return recyclerView;
    }

    abstract TwoLinedWithLink[] getData();
}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/links/LinkListFragment.kt
package org.ligi.gobandroid_hd.ui.links

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import org.ligi.gobandroid_hd.R

abstract class LinkListFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?)
            = (inflater.inflate(R.layout.recycler_view, container, false) as RecyclerView).apply {
        adapter = TwoLineRecyclerAdapter(getData())
        layoutManager = LinearLayoutManager(container!!.context)

    }

    internal abstract fun getData(): Array<LinkWithDescription>
}
